tools/pubsys/src/repo.rs

```diff
@@ -374,17 +369,32 @@ where
         // If we fail to load, but we only failed because the repo doesn't exist yet, then start
         // fresh by signalling that there is no known repo.  Otherwise, fail hard.
         Err(e) => {
-            if transport.repo_not_found.get() {
+            if is_file_not_found_error(&e) {
                 Ok(None)
             } else {
                 Err(e).with_context(|| error::RepoLoad {
                     metadata_base_url: metadata_url.clone(),
-                })?
+                })
             }
         }
     }
 }
 
+/// Inspects the `tough` error to see if it is a `Transport` error, and if so, is it `FileNotFound`.
+fn is_file_not_found_error(e: &tough::error::Error) -> bool {
+    if let tough::error::Error::Transport {
+        url: _,
+        source,
+        backtrace: _,
+    } = e
```

Comment:
```suggestion
    if let tough::error::Error::Transport { source, .. } = e
```
I think that'll work.